compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3396,6 +3398,40 @@ object Types extends TypeUtils {
     }
   }
 
+  // --- FlexibleType -----------------------------------------------------------------
+
+  /* Represents a nullable type coming from Java code in a similar way to Platform Types
+   * in Kotlin. A FlexibleType(T) generally behaves like an abstract type with bad bounds
+   * T|Null .. T, so that T|Null <: FlexibleType(T) <: T.
+   */
+  case class FlexibleType(original: Type, lo: Type, hi: Type) extends CachedProxyType with ValueType {
```

Comment:
Could [`FromJavaObject`](https://github.com/lampepfl/dotty/blob/85423424a990663b247d594dc632a749e316f1a0/compiler/src/dotty/tools/dotc/core/Definitions.scala#L331-L424) be replaced by a `FlexibleType(original = Object, lo = Any, hi = Object)` ?

Reply:
Yes, I think it would be interesting to separate the `FlexibleType` as a standalone tool for "special" subtyping, and do an experiment with `ObjectFromJava`.

Reply:
Note that references to `<special-ops>.FromJavaObject` exist in published TASTy files. So it still needs to exist as a type alias. Its alias might be a `FlexibleType`.

Erasure, in particular in the presence of `Array`s, would have to be carefully assessed.

Reply:
I did some experiment. I can't just replace the definition of `FromJavaObject` to a `FlexibleType` directly.
Since every `Object` from Java is a `FromJavaObject`, the `Object`s at class definition places are required to be a class type, but `FlexibleType` is not.

Reply:
I don't understand why we make `Object`s outside type signatures `FromJava` as well, as they should not use the special subtyping rule.

So, I think it is still possible to make `FromJavaObject ` flexible.